providers/implementations/exchange/build.info

```diff
@@ -1,13 +1,12 @@
 # We make separate GOAL variables for each algorithm, to make it easy to
 # switch each to the Legacy provider when needed.
 
-$DH_GOAL=../../libimplementations.a
 $ECX_GOAL=../../libimplementations.a
-$ECDH_GOAL=../../libimplementations.a
 $KDF_GOAL=../../libimplementations.a
 
 IF[{- !$disabled{dh} -}]
-  SOURCE[$DH_GOAL]=dh_exch.c
+  SOURCE[../../libfips.a]=dh_exch.c
+  SOURCE[../../libnonfips.a]=dh_exch.c
```

Comment:
[sigh] we really shouldn't do this, but rather have separate sources with "FIPS helper functions" and leave the implementations per se in libimplementations.a...  One on the list of my mental notes to deal with...

Reply:
please show me what you mean in this case..

Reply:
ok after re-reading I know what you mean..

Reply:
Reworked to remove your sigh.. (Took a while).